lib/ASTGen/Sources/ASTGen/ParameterClause.swift

```diff
@@ -0,0 +1,111 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import CASTBridging
+import SwiftSyntax
+
+// MARK: - ParamDecl
+
+fileprivate protocol ValueParameterSyntax: SyntaxProtocol {
+  var _firstName: TokenSyntax? { get }
+
+  var secondName: TokenSyntax? { get }
+
+  var _type: TypeSyntax? { get }
+
+  var defaultArgument: InitializerClauseSyntax? { get }
+}
+
+extension FunctionParameterSyntax: ValueParameterSyntax {
+  fileprivate var _firstName: TokenSyntax? {
+    firstName
+  }
+
+  fileprivate var _type: TypeSyntax? {
+    type
+  }
+}
+
+extension EnumCaseParameterSyntax: ValueParameterSyntax {
+  fileprivate var _firstName: TokenSyntax? {
+    firstName
+  }
+
+  fileprivate var _type: TypeSyntax? {
+    type
+  }
+}
+
+extension ASTGenVisitor {
+  private func makeParamDecl(_ node: some ValueParameterSyntax) -> ASTNode {
+    // FIXME: This location should be derived from the type repr.
+    let specifierLoc: BridgedSourceLoc = nil
+
+    let firstName: BridgedIdentifier
+    if let name = node._firstName {
+      if name.text == "_" {
```

Comment:
Comparing the `tokenKind` is much cleaner than going to the text-representation.

```suggestion
      if name.tokenKind == .wildcard {
```